en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,225 @@
+If you run into a build error telling you that you have gone above the maximum flash memory, you will need to disable other features in the `.px4board` file you are working on, or in the `default.px4board` file.
+Be careful not to disable something you need.
+
+For example, if you found you were running out of memory on FMUv4 boards you could disable SIH mode by setting `CONFIG_MODULES_SIMULATION_SIMULATOR_SIH=n` in [boards/px4/fmu-v4/default.px4board](https://github.com/PX4/PX4-Autopilot/blob/main/boards/px4/fmu-v4/default.px4board#L76), which may free up enough flash memory to allow crypto to be added.
+:::
+
+#### Pixhawk FMUv5 boards
+
+The FMUv5 board already has a custom make target `px4-fmu-v5_cryptotest` that you can use to build custom firmware with the required modules and "test" RSA keys.
+The configuration file that enables the above make target is [`cryptotest.px4board`](https://github.com/PX4/PX4-Autopilot/blob/main/boards/px4/fmu-v5/cryptotest.px4board) file in `boards/px4/fmu-v5`.
+The relevant keys in that file are reproduced below:
+
+```config
+CONFIG_BOARD_CRYPTO=y
+CONFIG_DRIVERS_STUB_KEYSTORE=y
+CONFIG_DRIVERS_SW_CRYPTO=y
+CONFIG_PUBLIC_KEY0="../../../Tools/test_keys/key0.pub"
+CONFIG_PUBLIC_KEY1="../../../Tools/test_keys/rsa2048.pub"
+```
+
+::: details Overview of the keys
+
+| Argument                     | Description                                                                                                                                                                                     |
+| ---------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| CONFIG_BOARD_CRYPTO          | Include crypto module in firmware.<br />= `y`: Enable log encryption.<br />= `n`: Disable log encryption.                                                                                       |
+| CONFIG_DRIVERS_SW_CRYPTO     | Include the built-in PX4 crypto backend library (used by above library).<br />= `y`: Enable<br />= `n`: Disable                                                                                 |
+| CONFIG_DRIVERS_STUB_KEYSTORE | Includes the built-in PX4 stub keystore driver.<br />= `y`: Enable<br />= `n`: Disable                                                                                                          |
+| CONFIG_PUBLIC_KEY0           | Location of public key for keystore index 0. By default, this is set as the exchange key by [SDLOG_EXCH_KEY](../advanced_config/parameter_reference.md#SDLOG_EXCH_KEY).<br />= `{path to key0}` |
+| CONFIG_PUBLIC_KEY1           | Location of public key for keystore index 1. By default, this is set as the encryption key by [SDLOG_KEY](../advanced_config/parameter_reference.md#SDLOG_KEY).<br />= `{path to key1}`         |
```

Comment:
```suggestion
| CONFIG_PUBLIC_KEY0           | Location of public key for keystore index 0.  |
| CONFIG_PUBLIC_KEY1           | Location of public key for keystore index 1. By default, this is set as the exchange key by [SDLOG_EXCH_KEY](../advanced_config/parameter_reference.md#SDLOG_EXCH_KEY).<br />= `{path to key1}`        |
```

I believe my first suggestion had it wrong. Since both SDLOG_KEY, SDLOG_EXCH_KEY and the CONFIG_PUBLIC_KEY params are zero indexed, I believe CONFIG_PUBLIC_KEY1 maps to SDLOG_EXCH_KEY by default. CONFIG_PUBLIC_KEY0 would map to nothing.

(SDLOG_EXCH_KEY defaults to 1. SDLOG_KEY defaults to 2)